### PR TITLE
chore(ci): label security tests and generate summary report COMPASS-7960

### PR DESCRIPTION
**Pull request by addaleax**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass/src/main/auto-connect.spec.ts [hunk 1]
```diff
@@ -20,6 +20,15 @@ const URI_WITH_DISALLOWED_SERVICE_HOST_AUTH_MECHANISM_PROP =
 const URI_WITH_ALLOWED_SERVICE_HOST_AUTH_MECHANISM_PROP =
   'mongodb://localhost:27017/?authMechanism=GSSAPI&authSource=%24external&authMechanismProperties=CANONICALIZE_HOST_NAME%3Aforward';
 
+/**
+ * @securityTests Automatic Connection Establishment Tests
```
Comment (mabaasit) [inline packages/compass/src/main/auto-connect.spec.ts:24]: additional _s_?

File: packages/compass-export-to-language/src/modules/transpiler.spec.ts [hunk 2]
```diff
@@ -154,6 +154,7 @@ FindIterable<Document> result = collection.find(filter);`);
     });
 
     for (const protectConnectionStrings of [false, true]) {
+      // Security-relevant test -- description is in the protect-connection-strings e2e test.
```
Comment (mabaasit) [inline packages/compass-export-to-language/src/modules/transpiler.spec.ts:157]: these are not used anywhere yet, right?

Comment (addaleax) [reply packages/compass-export-to-language/src/modules/transpiler.spec.ts:157]: Yup – just to clarify that a) these are relevant to security and b) already included in the report elsewhere

File: packages/compass/src/main/auto-connect.spec.ts (hunk 1 above)
Comment (addaleax) [reply packages/compass/src/main/auto-connect.spec.ts:24]: Yup, thanks for catching!

```suggestion
 * @securityTest Automatic Connection Establishment Tests
```

**Review by mabaasit — APPROVED** [review]

**Pull request merged**